lib/commands/general.js

```diff
@@ -201,12 +201,27 @@ helpers.suspendChromedriverProxy = function () {
   this.jwpProxyActive = true;
 };
 
+/**
+ * @typedef {Object} DeviceInfo
+ *
+ * @property {Object} * The result of '/appium/device/info' by UIAutomator2 server
+ * @property {?string} currentLocale - Current device locale. e.g. 'ja_EN' and 'zh-Hant_US'
+ * @property {?string} timeZone - TZ database Time Zones format like 'US/Pacific'
+ */
+
 /**
  * The list of available info entries can be found at
- * https://github.com/appium/appium-uiautomator2-server/blob/master/app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java
+ * https://github.com/appium/appium-uiautomator2-server/blob/master/app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java,
+ * device locale and timezone
+ *
+ * @returns {DeviceInfo} The current device info
+ * @throws {Error} if an error raised by command
  */
 commands.mobileGetDeviceInfo = async function () {
-  return await this.uiautomator2.jwproxy.command('/appium/device/info', 'GET');
+  const info = await this.uiautomator2.jwproxy.command('/appium/device/info', 'GET');
+  info.locale = await this.adb.getDeviceLocale();
```

Comment:
adb calls are expensive. I'd rather gather this info in the server code

Reply:
reasonable. Will add them for both espresso and uia2